### DISCUSSION TITLE
Need help configuring and installing...

### DISCUSSION THREAD

**Discussion by caseystive** [devcompl/TTController#99] [Configuration] [answered]
Hey all...

I'm having serious issues with my riing trio fans especially when using TT RGB Plus so I thought I would try out this TT Controller; however, I'm not the greatest with coding and all that so I could definitely use some help.

Moshi if we could quickly talk (phone/discord) and I could get a walkthrough, I'd very much appreciate it! 

Thanks for the help!

- Casey

**Comment by MoshiMoshi0** [comment]
Please post the output of `Main Menu -> Debug -> Report`

**Comment by caseystive** [comment]
```
Info
-------------------------------
»
»        OS: Windows 10 Enterprise 2009 [19041.1.amd64fre.vb_release.191206-1406]
»        Build: 0.11.0-beta+Branch.master.Sha.42768ba987b22e18ff8cdd114f754e5f7ce16d72
»
-------------------------------
Applications
-------------------------------
»
»
-------------------------------
HID
-------------------------------
»
»  [0x264a, 0x2260]: \\?\hid#vid_264a&pid_2260#9&33625e65&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
»
-------------------------------
Controllers
-------------------------------
»
»  Name: Riing Quad
»        VendorId: 9802
»        ProductId: 8800
»        Version: 3.0.3
»        Ports:
»                1:
»                        Data: [PortId: 1, Speed: 60, Rpm: 969, Unknown: 255]
»                        Identifier: [9802, 8800, 1]
»                2:
»                        Data: [PortId: 2, Speed: 60, Rpm: 979, Unknown: 255]
»                        Identifier: [9802, 8800, 2]
»                3:
»                        Data: [PortId: 3, Speed: 60, Rpm: 963, Unknown: 255]
»                        Identifier: [9802, 8800, 3]
»                4:
»                        Data: [PortId: 4, Speed: 60, Rpm: 0, Unknown: 255]
»                        Identifier: [9802, 8800, 4]
»                5:
»                        Data: [PortId: 5, Speed: 60, Rpm: 0, Unknown: 255]
»                        Identifier: [9802, 8800, 5]
»        Available effect types: PerLed
»
-------------------------------
Sensors
-------------------------------
»
»  Nuvoton NCT6798D:
»        Temperature:
»                Temperature #1 (/lpc/nct6798d/temperature/1): 26.5 °C
»                Temperature #2 (/lpc/nct6798d/temperature/2): 23.0 °C
»                Temperature #3 (/lpc/nct6798d/temperature/3): 95.0 °C
»                Temperature #4 (/lpc/nct6798d/temperature/4): 23.0 °C
»                Temperature #5 (/lpc/nct6798d/temperature/5): 23.0 °C
»                Temperature #6 (/lpc/nct6798d/temperature/6): 97.0 °C
»
»  AMD Ryzen 7 5800X:
»        Temperature:
»                Core (Tctl/Tdie) (/amdcpu/0/temperature/2): 47.6 °C
»                CCD1 (Tdie) (/amdcpu/0/temperature/3): 47.0 °C
»
»  NVIDIA GeForce RTX 3090:
»        Temperature:
»                GPU Core (/gpu-nvidia/0/temperature/0): 24.0 °C
»
»  Samsung SSD 970 EVO Plus 2TB:
»        Temperature:
»                Temperature (/nvme/0/temperature/0): 33.0 °C
»                Temperature 1 (/nvme/0/temperature/6): 33.0 °C
»                Temperature 2 (/nvme/0/temperature/7): 34.0 °C
»
-------------------------------
Press any key to continue...
```


~This is all I see so far...my main goal is to get something simple set up to accomplish these goals (again this is with 3 riing trio fans even though it seems to think I have riing quad fans):
1: The fans start as soon as possible (when I press power or whenever you would suggest coding them to start)
2: The fans should be straight red (255,0,0)
3: They ramp up and down depending on temps but only a few percentage every 2.5 or so seconds in order to avoid spikes in temps and voltages

That's really all I am trying to configure but I'm really confused how this works.

I could really use your guys help since I'm almost afraid to use my comp before I get these fans programmed correctly,  and as you can see I'm also dealing with some pretty high end hardware 😆

I've seen many great things about your controller online and hope it'll work well for me too.

I'd also be willing to donate over paypal towards your cause if this ends up working how I've heard it does.

Also again if we could somehow talk to make things go faster and in case I have more questions that'd be amazing since I feel as if using github to type back and forth might hinder things a little...but whatever works for you. :)

Thanks again!

- Casey

**Comment by MoshiMoshi0** [comment, marked as answer]
> again this is with 3 riing trio fans even though it seems to think I have riing quad fans

I think thermaltake started including riing quad controllers with older fans.

Try this config:

You have to tweak the pwm `"CurvePoints"` to fit your pc, the left value is temperature, right value is fan speed.
And set the `"CriticalValue"` value for the sensor.

Otherwise it should work like you want.

```json
{
  "Profiles": [
    {
      "Name": "Default",
      "Ports": [
        [9802, 8800, 1],
        [9802, 8800, 2],
        [9802, 8800, 3]
      ],
      "SpeedControllers": [
        {
          "Type": "PwmSpeedController", 
          "Config": {
            "CurvePoints": [
              [20, 30],
              [30, 50],
              [50, 60],
              [55, 75],
              [65, 100]
            ],
            "Sensors": ["/amdcpu/0/temperature/2"],
            "MinimumChange": 4,
            "MaximumChange": 8,
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ],
      "Effects": [
        {
          "Type": "StaticColorEffect",
          "Config": {
            "Color": {
              "Full": [255, 0, 0]
            },
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ]
    }
  ],
  "ComputerStateProfiles": [
    {
      "StateType": "Shutdown",
      "Ports": [
        [9802, 8800, 1],
        [9802, 8800, 2],
        [9802, 8800, 3]
      ],
      "Speed": 35
    },
    {
      "StateType": "Boot",
      "Ports": [
        [9802, 8800, 1],
        [9802, 8800, 2],
        [9802, 8800, 3]
      ],
      "Speed": 35
    }
  ],
  "PortConfigs": [
    {
      "Ports": [
        [9802, 8800, 1],
        [9802, 8800, 2],
        [9802, 8800, 3]
      ],
      "Config": {
        "DeviceType": "RiingTrio",
        "IgnoreColorCache": true
      }
    }
  ],
  "SensorConfigs": [
    {
      "Sensors": ["/amdcpu/0/temperature/2"],
      "Config": {
        "CriticalValue": 80
      }
    }
  ]
}
```

**Reply by caseystive** [reply]
Thank you so much for the quick response!

Just a few other quick questions:
1. Should I be moving the folder with all the config to a location where it'll stay there forever...I thought I remember seeing somewhere that you can't move it while it's installed and right now the folder is just sitting in my downloads folder...?
2. Where should I be pasting this code into  / under? Does it all just paste into one location or do I have to split up where I place each section?
3. Is it also ok that the service seems to think that I'm on Windows Enterprise when I'm actually on Windows 10 Pro (probably doesn't matter, just wanted to confirm)

Thank you!

**Comment by MoshiMoshi0** [comment]
> Should I be moving the folder with all the config to a location where it'll stay there forever...I thought I remember seeing somewhere that you can't move it while it's installed and right now the folder is just sitting in my downloads folder...?

Yes once you install it from the menu you cant move the files, but its always easy to uninstall, move the files and install again.
You can use the console mode for running without installing.

Should be explained pretty well on the [docs](https://moshimoshi0.github.io/TTController/).

> Where should I be pasting this code into / under? Does it all just paste into one location or do I have to split up where I place each section?

All into `config.json` file that is in the same directory as service executable.

> Is it also ok that the service seems to think that I'm on Windows Enterprise when I'm actually on Windows 10 Pro (probably doesn't matter, just wanted to confirm)

Does not matter.

**Reply by caseystive** [reply]
Ok perfect!

Lastly...do I need a specific program / software to edit the JSON file because it won't let me open it?

**Reply by MoshiMoshi0** [reply]
Oh, no just open it in any text editor, notepad will do.
Should probably mention that in the documentation.

**Reply by caseystive** [reply]
Oh ok cool thats easy. I appreciate the help again!

I'm going to go try all those configurations and I'll get back to you if it doesn't work. Hopefully it does though! :)

**Reply by caseystive** [reply]
Also the "critical value" you told me to set...what does that do / tell the system?

**Reply by MoshiMoshi0** [reply]
The `PwmSpeedController` will slowly adjust the fan speed so its not erratic, that might cause the fan speed to take a while to ramp up if you start doing super heavy workloads from idle.
If the temperature gets to that critical value it will override all speed controllers and set all fans to 100%.

**Reply by caseystive** [reply]
Hey Moshi,

I got everything configured it seems to be working great! I really appreciate the help...I sent you a little something on paypal as thanks.

One last thing is it does still seem like the fans are ramping up a little fast.

I know that the PwmSpeedController tells it _what_ speed to ramp to when at a certain temperature...but what tells it _how_ fast to ramp up to that speed? For instance I still feel as if it's ramping up to those speeds almost immediately. If I wanted it to take around 6 seconds to ramp from %30 to %50 how would I do that?

**Reply by MoshiMoshi0** [reply]
With current settings it cant change speeds almost immediately. It should take 7.5s to go from 30 to 50, it only changes speed every 2.5s.

[MinimumChange](https://moshimoshi0.github.io/TTController/plugins/speed-controllers/pwmspeedcontroller#minimumchange) and [MaximumChange](https://moshimoshi0.github.io/TTController/plugins/speed-controllers/pwmspeedcontroller#maximumchange) set the limits if you want to tweak those.
